internal/app/orchestrator/downstream.go

```diff
@@ -14,82 +14,73 @@ package orchestrator
 import (
 	"sync"
 
-	"github.com/envoyproxy/xds-relay/internal/app/mapper"
 	"github.com/envoyproxy/xds-relay/internal/app/transport"
 )
 
 // downstreamResponseMap is a map of downstream xDS client requests to response
 // channels.
 type downstreamResponseMap struct {
 	mu      sync.RWMutex
-	watches map[transport.Request]transport.Watch
+	watches map[string][]transport.Watch
```

Comment:
For performance reasons, I think it's ok we sign off on this approach. My only concern is we lose some debug information that comes from the having the request available.  Ex: connected nodes to a particular key. But perhaps this is available in the watch?

Reply:
Yes. The tradeoff here is to keep the core path simple...we should add the deuggability feature as a map as long as we are not looping over the map entries.